src/components/widgets/accounts-expense-weight/accounts-expense-weight.directive.coffee

```diff
@@ -42,9 +42,11 @@ module.controller('WidgetAccountsExpenseWeightCtrl', ($scope, $q, ChartFormatter
       else
         $translate.instant("impac.widget.account_expense_weight.comparator.total_exp")
 
-  $scope.displayAccount = ->
-    $scope.updateSettings(false).then ->
-      w.format()
+  $timeout ->
```

Comment:
why do you need a `$timeout` here?
the `timeout` should be applied to the execution of `w.format()`, not to the definition of the method (alos, `, 0` is optional:
```coffeescript
$scope.displayAccount = ->
  $scope.updateSettings(false).then ->
    $timeout(-> w.format())
```